### PR TITLE
Preserve modified date range while DetailView is displayed.

### LINKED ISSUE #623: X-Range should persist across selection
Re: #619

When using the large time series chart, one can use the date range widget to change the range of the plot:

image

Currently if you add another area to the selection, this date range gets reset, causing the user to have to navigate back to the area they found interesting.

We should persist the current date range across geo selections to improve this behavior.

### PR DESCRIPTION
**Pull request by dlaliberte**
closes #623

**Prerequisites**:

- [x] Unless it is a hotfix it should be merged against the `dev` branch
- [x] Branch is up-to-date with the branch to be merged with, i.e. `dev`
- [x] Build is successful
- [x] Code is cleaned up and formatted

### Summary

Preserve the modified date range in the DetailView between changes of sensor or regions. Also entering and exiting compare mode leave the date range as it is. Only closing the DetailView restores the default, next time it is opened.

### CODE REVIEW

**Comment by dlaliberte** [comment]
It actually fails to preserve the date range the first time after opening the DetailView.  And if you select a sensor with no data, it loses the date range entirely.

**Comment by dlaliberte** [comment]
Fixed the "no data" situation.  The failure to preserve the date on the first time doesn't happen if I am watching with the debugger, so there is a timing issue or something that I am not understanding.

**Comment by dlaliberte** [comment]
Redefined the dateRange variable to use writable, but it still fails (most of the time) the on the first change, but works thereafter.  I'm stumped.

File: src/components/DetailView/DetailView.svelte
Comment (sgratzl) [inline src/components/DetailView/DetailView.svelte:88]: can you remove the log statement, please

File: src/components/DetailView/DetailView.svelte
Comment (sgratzl) [inline src/components/DetailView/DetailView.svelte:94]: same here

**Review by sgratzl — COMMENTED** [review]
> Redefined the dateRange variable to use writable, but it still fails (most of the time) the on the first change, but works thereafter. I'm stumped.

how can I reproduce that? can you make a screenshot showing the behavior?

File: src/components/DetailView/DetailView.svelte
Comment (dlaliberte) [reply src/components/DetailView/DetailView.svelte:88]: Yes, well, my thought was that it helps to have the log messages while trying to debug why it fails the first time.  I'll remove them before this is ready to merge.

File: src/components/DetailView/DetailView.svelte
Comment (dlaliberte) [reply src/components/DetailView/DetailView.svelte:94]: ack.  will do.  

BTW I don't think the console messages make the problem worse, though that might be possible.

File: src/components/DetailView/DetailView.svelte
Comment (sgratzl) [reply src/components/DetailView/DetailView.svelte:88]: then please convert this PR to a draft such that I know it is not ready to be reviewed/merged. To my understanding if I'm assigned as a reviewer and the PR is not a draft, I assume I should review it because it is in its final state

**Comment by dlaliberte** [comment]
> how can I reproduce that? can you make a screenshot showing the behavior?

I'm not sure if I can do a screen recording.  Need a tool allowed by Google that works with macbook and/or chrome.

But reproducing this is straightforward.  On a fresh page (I click on "COVIDcast" link in header), 
- starting with Map Overview, 
- click on magnifier icon for COVID-Related Doctor Visits, 
- drag the date range slider back a couple months, 
- click the magnifier for COVID-like Symptoms
- Notice that the date range slider is reset to the small multiple default.
- Try it again (without restarting), and notice it works correctly thereafter.

Actually, the latest version preserves the change after closing the detail view as well, or going to any other view.  That's because it is storing it in detailViewTimeSpan, so it never gets reset to the default after that.  It only fails the very first time.

**Comment by dlaliberte** [comment]
> Actually, the latest version preserves the change after closing the detail view as well, or going to any other view. That's because it is storing it in detailViewTimeSpan, so it never gets reset to the default after that. It only fails the very first time.

After merging in the latest from origin/dev, now it does work the first time as well.  So something has changed maybe about the timing of page loading or whatever, so I wonder if this will be reliable.

Also, the persistence of the date range change could be a useful feature, but now we have the opposite problem, that there is no way to reset to the default, other than refreshing the page.   I'll remove it from the store, and see if that works correctly now.

File: src/components/DetailView/DetailView.svelte
Comment (dlaliberte) [reply src/components/DetailView/DetailView.svelte]: All works correctly now.

File: src/components/DetailView/DetailView.svelte
Comment (dlaliberte) [reply src/components/DetailView/DetailView.svelte]: I needed your feedback (which was useful, by the way), and assumed you would want to see the console messages as well.  

But I think the 'draft' status is not quite right, unless I mean to say "you can ignore this PR until further notice", which is not what I wanted to convey.   We would need another status that indicates "Not ready to submit, but your feedback is needed".   And regardless, even if it is not in 'draft' mode, the PR would not be ready to submit until approved, so it means the same thing.

File: src/stores/index.js
Comment (sgratzl) [inline src/stores/index.js:136]: do we still need that?

**Review by sgratzl — APPROVED** [review]
looks good beside the one comment regarding the store

File: src/stores/index.js
Comment (dlaliberte) [reply src/stores/index.js:136]: Nope.  I forget to remove that.

File: src/components/DetailView/DetailView.svelte
Comment (dlaliberte) [inline src/components/DetailView/DetailView.svelte:85]: Maybe we shouldn't bother checking whether the value has changed, since it won't be used immediately anyway.

But the check that both values are not NaN is important anyway, since NaNs will be sent when there is no data for the sensor.

**Pull request merged**
